p2p/protocol/identify/obsaddr_test.go

```diff
@@ -19,110 +23,55 @@ import (
-		h.t.Fatal(err)
-	}
+var _ network.Conn = &mockConn{}
 
-	c = mocknet.ConnComplement(c)
-	if c.Stat().Direction != network.DirInbound {
-		h.t.Fatal("expected conn direction to be inbound")
+func newMockConn(dir network.Direction, peer peer.ID, local, remote ma.Multiaddr) network.Conn {
+	return &mockConn{
+		direction: dir,
+		peer:      peer,
+		local:     local,
+		remote:    remote,
 	}
-	return c
 }
 
-func (h *harness) observe(observed ma.Multiaddr, observer peer.ID) network.Conn {
-	c := h.conn(observer)
-	h.oas.Record(c, observed)
-	time.Sleep(50 * time.Millisecond) // let the worker run
-	return c
+func (m *mockConn) Close() error                  { panic("implement me") }
+func (m *mockConn) LocalPeer() peer.ID            { panic("implement me") }
+func (m *mockConn) LocalPrivateKey() ic.PrivKey   { panic("implement me") }
+func (m *mockConn) RemotePeer() peer.ID           { return m.peer }
+func (m *mockConn) RemotePublicKey() ic.PubKey    { panic("implement me") }
+func (m *mockConn) LocalMultiaddr() ma.Multiaddr  { return m.local }
+func (m *mockConn) RemoteMultiaddr() ma.Multiaddr { return m.remote }
+func (m *mockConn) Stat() network.ConnStats {
+	return network.ConnStats{Stats: network.Stats{Direction: m.direction}}
 }
+func (m *mockConn) Scope() network.ConnScope                              { panic("implement me") }
+func (m *mockConn) ID() string                                            { panic("implement me") }
+func (m *mockConn) NewStream(ctx context.Context) (network.Stream, error) { panic("implement me") }
+func (m *mockConn) GetStreams() []network.Stream                          { panic("implement me") }
 
-func (h *harness) observeInbound(observed ma.Multiaddr, observer peer.ID) network.Conn {
-	c := h.connInbound(observer)
-	h.oas.Record(c, observed)
-	time.Sleep(50 * time.Millisecond) // let the worker run
-	return c
-}
-
-func newHarness(t *testing.T) harness {
-	mn := mocknet.New()
+func newPeer(t *testing.T) peer.ID {
+	t.Helper()
 	sk, _, err := ic.GenerateECDSAKeyPair(rand.Reader)
 	require.NoError(t, err)
-	h, err := mn.AddPeer(sk, ma.StringCast("/ip4/127.0.0.1/tcp/10086"))
-	require.NoError(t, err)
-	oas, err := identify.NewObservedAddrManager(h)
+	id, err := peer.IDFromPrivateKey(sk)
 	require.NoError(t, err)
-	t.Cleanup(func() {
-		mn.Close()
-		oas.Close()
-	})
-	return harness{
-		oas:     oas,
-		mocknet: mn,
-		host:    h,
-		t:       t,
-	}
+	return id
 }
 
-// TestObsAddrSet
 func TestObsAddrSet(t *testing.T) {
-	addrsMatch := func(a, b []ma.Multiaddr) bool {
-		if len(a) != len(b) {
-			return false
-		}
-		for _, aa := range a {
-			var found bool
-			for _, bb := range b {
-				if aa.Equal(bb) {
-					found = true
-					break
-				}
-			}
-			if !found {
-				return false
-			}
-		}
-		return true
-	}
+	local := ma.StringCast("/ip4/127.0.0.1/tcp/10086")
+	h, err := libp2p.New(libp2p.ListenAddrs(local))
+	require.NoError(t, err)
+	defer h.Close()
+
+	oam, err := identify.NewObservedAddrManager(h)
```

Comment:
Is it possible to see some flakiness because the number of observations becomes greater than the buffer size for the [channel](https://github.com/libp2p/go-libp2p/blob/identify-test-no-more-mocknet/p2p/protocol/identify/obsaddr.go#L135)? Also I'm concerned with the dropping log being just a [debug](https://github.com/libp2p/go-libp2p/blob/identify-test-no-more-mocknet/p2p/protocol/identify/obsaddr.go#L240), maybe a warn would be more appropriate?